### ISSUE TITLE
Support for MongoDB indexes 

### ISSUE THREAD

**Issue by Davide-Gheri** [mikro-orm/mikro-orm#159]
**Is your feature request related to a problem? Please describe.**
I just started exploring this library and I have to say, I like it.
Just one thing, I was expecting indexes for mongodb working just like in other libraries (TypeORM, mongoose, Typegoose) but I didn't find anything in the docs and no implementation in the code.
Right now I found a workaround by manually call the underlying Mongodb connection:
```typescript
const connection = await MikroORM.init({
  //...options
});

connection.em.getConnection<MongoConnection>()
  .getCollection('user').createIndex({email: 1}, {unique: true, name: 'email'});
```
But gets messy when you have lots of entities or a "modular" application (ex. Nestjs)

**Describe the solution you'd like**
it would be cool to automatically add indexes on startup, adding an `index` options on `@Property` or creating a new `@Index` decorator that optionally receives Mongodb IndexOptions object and the index type (es 2dsphere)

**Comment by B4nan** [comment]
Sure thing, I wanted to add support for that in sql drivers as well, so will try to come up with unified way of doing this across all drivers. Not 100% sure if we want to trigger that during initialization though... Will need to think about it a bit more, we might as well use the CLI command for schema update, or maybe add new one like `ensure-indexes` as whole schema generator is about sql only.

**Comment by osayfun** [comment]
Can you also plan for supporting `@Unique` decorator for MongoDB?

**Comment by B4nan** [comment]
Sure

**Comment by ttsirkia** [comment]
Mongoose handles this rather nicely: https://mongoosejs.com/docs/guide.html#indexes By default, indexes are automatically generated but this can be disabled which is suggested in production environment.

**Comment by osayfun** [comment]
> Mongoose handles this rather nicely: https://mongoosejs.com/docs/guide.html#indexes By default, indexes are automatically generated but this can be disabled which is suggested in production environment.

I think you reply is irrelevant here

**Comment by ttsirkia** [comment]
@osayfun may I ask why? There was discussion above how this could be implemented and should it be a separate command, etc. I pointed out how this is implemented in another library.

**Comment by osayfun** [comment]
@ttsirkia Discussion is about the mikro-orm not supporting this feature right now. Mongodb indexes can be achieved in so many ways(even adding manually)

**Comment by ttsirkia** [comment]
Yes, but if they are supported and implemented at some point to this library, is assume that the following needs to be resolved in order to answer the question "how".

_Not 100% sure if we want to trigger that during initialization though... Will need to think about it a bit more, we might as well use the CLI command for schema update, or maybe add new one like ensure-indexes as whole schema generator is about sql only._

My comment was a reply to this. In Mongoose, `ensureIndexes` is called during start if not separately disabled.

**Comment by osayfun** [comment]
Oh right! Sorry, my bad.

**Comment by alidaghighi** [comment]
Any updates on this?

**Comment by ttsirkia** [comment]
Some information in this comment: https://github.com/mikro-orm/mikro-orm/issues/385#issuecomment-595766797

**Comment by B4nan** [comment]
Available in v3.4.0
